lib/spells/BattleSpellMechanics.cpp

```diff
@@ -629,17 +631,20 @@ void ForceFieldMechanics::setupObstacle(SpellCreatedObstacle * obstacle) const
 ///RemoveObstacleMechanics
 void RemoveObstacleMechanics::applyBattleEffects(const SpellCastEnvironment * env, const BattleSpellCastParameters & parameters, SpellCastContext & ctx) const
 {
-	if(auto obstacleToRemove = parameters.cb->battleGetObstacleOnPos(parameters.getFirstDestinationHex(), false))
+	auto obstacleToRemove = parameters.cb->battleGetAllObstaclesOnPos(parameters.getFirstDestinationHex(), false);
+	if(!obstacleToRemove.empty())
 	{
-		if(canRemove(obstacleToRemove.get(), parameters.spellLvl))
-		{
-			ObstaclesRemoved obr;
-			obr.obstacles.insert(obstacleToRemove->uniqueID);
-			env->sendAndApply(&obr);
-		}
-		else
-		{
-			env->complain("Cant remove this obstacle!");
+		for(auto &i : obstacleToRemove){
+			if(canRemove(i.get(), parameters.spellLvl))
+			{
+				ObstaclesRemoved obr;
+				obr.obstacles.insert(i->uniqueID);
+				env->sendAndApply(&obr);
+			}
+			else
+			{
+				env->complain("Cant remove this obstacle!");
```

Comment:
put all obstacles in one `ObstaclesRemoved` pack and complain only if `obr.obstacles` is empty